linux_os/guide/system/accounts/accounts-restrictions/password_storage/no_empty_passwords/ansible/shared.yml

```diff
@@ -1,14 +1,72 @@
-# platform = multi_platform_rhel,multi_platform_fedora,multi_platform_ol,multi_platform_rhv,multi_platform_sle
+# platform = multi_platform_rhel,multi_platform_fedora,multi_platform_ol,multi_platform_rhv
 # reboot = false
 # strategy = configure
 # complexity = low
 # disruption = medium
-- name: "Prevent Log In to Accounts With Empty Password - system-auth"
-  replace:
-    dest: /etc/pam.d/system-auth
-    regexp: 'nullok'
+- name: Check if system relies on authselect
+  ansible.builtin.stat:
+    path: /usr/bin/authselect
+  register: result_authselect_present
+
+- name: Check integrity of authselect current profile
+  ansible.builtin.command:
+    cmd: authselect check
+  register: result_authselect_check_cmd
+  ignore_errors: yes
+  when:
+    - result_authselect_present.stat.exists
+
+- name: Get authselect current profile
+  ansible.builtin.shell:
+    cmd: authselect current -r | awk '{ print $1 }'
+  register: result_authselect_profile
+  when:
+    - result_authselect_check_cmd is failed
+
+- name: Get authselect current features
+  ansible.builtin.shell:
+    cmd: authselect current | tail -n+3 | awk '{ print $2 }'
+  register: result_authselect_features
+  when:
+    - result_authselect_check_cmd is failed
+
+- name: Define the current authselect profile as a local fact
+  ansible.builtin.set_fact:
+    authselect_profile: "{{ result_authselect_profile.stdout }}"
+  when:
+    - result_authselect_check_cmd is failed
+    - result_authselect_profile | default(false) | bool
+
+- name: Restore the authselect profile if manually modified
+  ansible.builtin.command:
+    cmd: authselect select {{ authselect_profile }} --force
+  register: result_pam_authselect_restore_profile
+  when:
+    - result_authselect_check_cmd is failed
+    - result_authselect_profile | default(false) | bool
```

Comment:
Is it a debugging leftover? Won't the `default(false)` cause the when condition to always fail?

Reply:
I made a manual change to the pam file and the task was skipped.
```
ansible-playbook -i "rhel85," --tags no_empty_passwords build/ansible/rhel8-playbook-stig.yml 

PLAY [all] ***********************************************************************************************************

TASK [Gathering Facts] ***********************************************************************************************
ok: [rhel85]

TASK [Check if system relies on authselect] **************************************************************************
ok: [rhel85]

TASK [Check integrity of authselect current profile] *****************************************************************
fatal: [rhel85]: FAILED! => {"changed": true, "cmd": ["authselect", "check"], "delta": "0:00:00.006320", "end": "2021-11-08 09:38:18.257149", "msg": "non-zero return code", "rc": 3, "start": "2021-11-08 09:38:18.250829", "stderr": "[error] [/etc/authselect/system-auth] has unexpected content!", "stderr_lines": ["[error] [/etc/authselect/system-auth] has unexpected content!"], "stdout": "Current configuration is not valid. It was probably modified outside authselect.", "stdout_lines": ["Current configuration is not valid. It was probably modified outside authselect."]}
...ignoring

TASK [Get authselect current profile] ********************************************************************************
changed: [rhel85]

TASK [Get authselect current features] *******************************************************************************
changed: [rhel85]

TASK [Define the current authselect profile as a local fact] *********************************************************
skipping: [rhel85]

TASK [Restore the authselect profile if manually modified] ***********************************************************
skipping: [rhel85]

TASK [Restore the authselect features if manually modified] **********************************************************
[DEPRECATION WARNING]: evaluating 'result_pam_authselect_restore_profile' as a bare variable, this behaviour will go 
away and you might need to add |bool to the expression in the future. Also see CONDITIONAL_BARE_VARS configuration 
toggle. This feature will be removed in version 2.12. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
skipping: [rhel85] => (item=without-nullok) 

TASK [Ensure nullok property is not present via authselect tool] *****************************************************
fatal: [rhel85]: FAILED! => {"changed": true, "cmd": ["authselect", "enable-feature", "without-nullok"], "delta": "0:00:00.007492", "end": "2021-11-08 09:38:19.853987", "msg": "non-zero return code", "rc": 4, "start": "2021-11-08 09:38:19.846495", "stderr": "[error] [/etc/authselect/system-auth] has unexpected content!\n[error] Unexpected changes to the configuration were detected.\n[error] Refusing to activate profile unless those changes are removed or overwrite is requested.\nUnable to enable feature [17]: File exists", "stderr_lines": ["[error] [/etc/authselect/system-auth] has unexpected content!", "[error] Unexpected changes to the configuration were detected.", "[error] Refusing to activate profile unless those changes are removed or overwrite is requested.", "Unable to enable feature [17]: File exists"], "stdout": "", "stdout_lines": []}

PLAY RECAP ***********************************************************************************************************
rhel85                     : ok=5    changed=3    unreachable=0    failed=1    skipped=3    rescued=0    ignored=1   
```

Reply:
This approach related to the `[DEPRECATION WARNING]` mentioned there. Let me check this conditional again, since I tested this case before... : (

Reply:
Fixed. Thanks.